Restore/adapt link color functionality for manifest v3.

colossatr0n opened the pull request:
I've tried to keep the changeLinkColor content script as simple as possible so that background.js does the heavy lifting.
This will reduce duplicate code between options.js, background.js and changeLinkColor.js, since the former 2 can now contain imports, whereas content scripts cannot.

colossatr0n committed:
Restore/adapt link color functionality for manifest v3.

I've tried to keep the changeLinkColor content script as simple as possible so that background.js does the heavy lifting.
This will reduce duplicate code between options.js, background.js and changeLinkColor.js, since the former 2 can now contain imports, whereas content scripts cannot.

colossatr0n committed:
Change color of all text related to an achor.

colossatr0n committed:
Highlight links on dom update..maybe?

colossatr0n committed:
Link highlighting now works on infinite scroll websites.

colossatr0n committed:
Move content scripts into their own directory.

colossatr0n committed:
Remove comment.

colossatr0n committed:
Reprocess link colors when a tab is activated.

colossatr0n committed:
Color text of the anchor element itself, not just its children.

edoreld committed:
Merge pull request #20 from colossatr0n/manifest-v3-only

Convert project to manifest v3 (replacement PR).

edoreld commented:
I'm not sure if I want to add this functionality back, but nevertheless I tried to see what it looks like, because it might look better than I expected. However, I couldn't make it work. 

How to reproduce:
1. Go to https://github.com/edoreld/mark-as-read
2. Click on "from the Chrome Web Store"
3. Mark it as read
4. Go back to https://github.com/edoreld/mark-as-read
5. Refresh the page
6. "from the Chrome Web Store" link hasn't change color

edoreld committed:
upgrade icons to v3 format

this fixes the issue where the extension image doesn't appear on the extensions page

edoreld committed:
Revert "upgrade icons to v3 format"

This reverts commit 6f4f217d3cdbb2ab83bec3b65d8de719a8985c4f.

colossatr0n commented:
Did you add https://github.com on the options page?

It's working for me:
<img width="804" alt="image" src="https://user-images.githubusercontent.com/29556317/168089705-ba68c56b-46ce-423b-bd94-a0f53405f7ac.png">

Try changing the default color to red or something. Maybe the default blue just looks really similar to the original link color.

colossatr0n committed:
Merge branch 'master' into restore-link-highlight

colossatr0n commented:
Closing this PR as I have recreated it here: https://github.com/edoreld/mark-as-read/pull/32
